### ISSUE TITLE
AKS - How to create virtual nodes when provisioning cluster

### ISSUE THREAD

**Issue by andtii** [Azure/azure-libraries-for-net#547]
We need to be able to create clusters with Virtual Node support enabled. Cant find how to do it!

**Comment by yaohaizh** [comment]
Could you investigate whether we have support this feature?

**Comment by andtii** [comment]
@ChenTanyi  @yaohaizh Any progress here?

**Comment by ChenTanyi** [comment]
@andtii I haven't investigated it. Maybe I'll try to investigate it next sprint.

**Comment by ChenTanyi** [comment]
@andtii the creation for virtual nodes uses the aciConnectorLinux add on, it could be written as:
```csharp

            var config = new Dictionary<string, string>();
            config.Add("SubnetName", virtualnetName);
            var addonProfile = new ManagedClusterAddonProfile(true, config);

            var addonMap = new Dictionary<string, ManagedClusterAddonProfile>();
            addonMap.Add("aciConnectorLinux", addonProfile);

            var virtualNetwork = azure.Networks.Define(networkName)
                .WithRegion(region)
                .WithExistingResourceGroup(rgName)
                .WithAddressSpace("10.0.0.0/8")
                .WithSubnet(akssubnetName, "10.240.0.0/16")
                .WithSubnet(virtualnetName, "10.241.0.0/16")
                .Create();

            var aks = azure.KubernetesClusters.Define(aksName)
                .WithRegion(region)
                .WithExistingResourceGroup(rgName)
                .WithLatestVersion()
                .WithRootUsername(username)
                .WithSshKey(sshKey)
                .WithServicePrincipalClientId(credentials.ClientId)
                .WithServicePrincipalSecret(clientSecret)
                .DefineAgentPool(agentPoolName)
                    .WithVirtualMachineSize(Microsoft.Azure.Management.ContainerService.Fluent.Models.ContainerServiceVMSizeTypes.StandardD2V2)
                    .WithAgentPoolVirtualMachineCount(1)
                    .WithVirtualNetwork(virtualNetwork.Id, akssubnetName)
                    .Attach()
                .WithDnsPrefix(dnsPrefixName)
                .DefineNetworkProfile
                    .WithNetworkPlugin(NetworkPlugin.Azure)
                    .Attach()
                .WithAddOnProfiles(addonMap)
                .Apply();
```
we may discuss about whether we should implement it as fluent interface.

**Comment by yaohaizh** [comment]
Fixed in PR#914.
